general/src/reason.rs

```diff
@@ -0,0 +1,96 @@
+//! Contains the
+
+use serde::Serialize;
+
+use crate::filter::FilterStatKey;
+
+/// Taken from Sentry's outcome reasons for invalid outcomes.
+/// Unlike FilterStatKey these are serialized with snake_case (e.g. project_id)
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Hash)]
+#[serde(rename_all = "snake_case")]
+pub enum OutcomeInvalidReason {
+    Duplicate,
+    ProjectId,
+    AuthVersion,
+    AuthClient,
+    NoData,
+    TooLarge,
+    DisallowedMethod,
+    ContentType,
+    MultiProjectId,
+    MissingMinidumpUpload,
+    InvalidMinidump,
+    SecurityReportType,
+    SecurityReport,
+    Cors,
+
+    //new errors emitted by Semaphore (and not found in Sentry)
+    StorePayloadError,
```

Comment:
Can we split this into more concrete types:

```
InvalidJson,
UnknownLength,
InvalidCompression,
```

Reply:
ok, done